src/serac/physics/utilities/functional/domain_integral.hpp

```diff
@@ -591,19 +494,42 @@ class DomainIntegral {
     //
     // note: the qf_derivatives_ptr is copied by value to each lambda function below,
     //       to allow the evaluation kernel to pass derivative values to the gradient kernel
-    evaluation_ = [=, &data](const mfem::Vector& U, mfem::Vector& R) {
-      domain_integral::evaluation_kernel<geometry, test_space, trial_space, Q>(U, R, qf_derivatives.get(), J_, X_,
-                                                                               num_elements, qf, data);
-    };
-
-    gradient_ = [=](const mfem::Vector& dU, mfem::Vector& dR) {
-      domain_integral::gradient_kernel<geometry, test_space, trial_space, Q>(dU, dR, qf_derivatives.get(), J_,
-                                                                             num_elements);
-    };
-
-    gradient_mat_ = [=](mfem::Vector& K_e) {
-      gradient_matrix_kernel<geometry, test_space, trial_space, Q>(K_e, qf_derivatives.get(), J_, num_elements);
-    };
+
+    if constexpr (std::is_same_v<execution_policy, serac::cpu_policy>) {
+      evaluation_ = [=, &data](const mfem::Vector& U, mfem::Vector& R) {
+        domain_integral::evaluation_kernel<geometry, test_space, trial_space, Q>(U, R, qf_derivatives.get(), J_, X_,
+                                                                                 num_elements, qf, data);
+      };
+
+      gradient_ = [=](const mfem::Vector& dU, mfem::Vector& dR) {
+        domain_integral::gradient_kernel<geometry, test_space, trial_space, Q>(dU, dR, qf_derivatives.get(), J_,
+                                                                               num_elements);
+      };
+
+      gradient_mat_ = [=](mfem::Vector& K_e) {
+        gradient_matrix_kernel<geometry, test_space, trial_space, Q>(K_e, qf_derivatives.get(), J_, num_elements);
+      };
+    }
+
+#if defined(__CUDACC__)
+    if constexpr (std::is_same_v<execution_policy, serac::gpu_policy>) {
+      evaluation_ = [=](const mfem::Vector& U, mfem::Vector& R) {
+        serac::detail::ThreadExecutionConfiguration exec_config{.blocksize = 128};
```

Comment:
Please pull this magic number into a define with a reason why you chose it if possible

Reply:
Let me add some documentation, but I think the execution configuration launch needs to be refactored anyways, so I'm hesitant to add a define since this should get addressed soon anyways.